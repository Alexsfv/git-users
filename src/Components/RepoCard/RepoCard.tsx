import { observer } from 'mobx-react'
import React from 'react'
import { getTimeString } from '../../types/func/time'
import { GithubRepository } from '../../types/types'
import RepoCardInfo from './RepoCardInfo/RepoCardInfo'

type RepoCardProps = {
    repo: GithubRepository
}
const RepoCard: React.FC<RepoCardProps> = observer(({repo}) => {

    return (
        <div className="card mx-2 mb-3" style={{width: '18rem'}}>
            <div className="d-flex flex-column card-body">
                <h3 className="card-title text-success">{repo.name}</h3>

                    <RepoCardInfo
                        title={'Описание'}
                        data={repo.description}
                    />

                    <RepoCardInfo
                        title={'Язык'}
                        data={repo.language}
                    />

                    <RepoCardInfo
                        title={'Создан'}
                        data={getTimeString(repo.created_at)}
                    />

                    <RepoCardInfo
                        title={'Обновлен'}
                        data={getTimeString(repo.updated_at)}
                    />

                    <RepoCardInfo
                        title={'Forks'}
                        data={repo.forks}
                    />

                    <RepoCardInfo
                        title={'Watchers'}
                        data={repo.watchers}
                    />

                    <RepoCardInfo
                        title={'Open Issues'}
                        data={repo.open_issues}
                    />


                <a href={repo.html_url} className="btn btn-primary ml-auto mt-auto" target="_blank">Перейти к репозиторию</a>
            </div>
        </div>
    )
})

export default RepoCard