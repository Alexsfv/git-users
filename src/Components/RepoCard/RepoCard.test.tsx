import { shallow } from 'enzyme'
import React from 'react'
import RepoCard from './RepoCard'

const repo = {
    id: 55,
    name: 'name',
    html_url: 'https://google.com',
    description: 'descr',
    created_at: '111-111-111',
    updated_at: '111-111-111',
    language: 'JS',
    forks: 5,
    open_issues: 1,
    watchers: 5,
}

describe("RepoCard component", () => {
    it("should render RepoCard", () => {
        const component = shallow(<RepoCard repo={repo}/>)
        expect(component).toMatchSnapshot()
    })

    it("should contain link", () => {
        const component = shallow(<RepoCard repo={repo}/>)
        const link = component.find('a')
        expect(link).toHaveLength(1)
    })
})