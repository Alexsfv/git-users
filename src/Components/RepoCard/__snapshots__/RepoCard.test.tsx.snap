// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoCard component should render RepoCard 1`] = `
<div
  className="card mx-2 mb-3"
  style={
    Object {
      "width": "18rem",
    }
  }
>
  <div
    className="d-flex flex-column card-body"
  >
    <h3
      className="card-title text-success"
    >
      name
    </h3>
    <Memo(wrappedComponent)
      data="descr"
      title="Описание"
    />
    <Memo(wrappedComponent)
      data="JS"
      title="Язык"
    />
    <Memo(wrappedComponent)
      data="NaN:NaN:NaN"
      title="Создан"
    />
    <Memo(wrappedComponent)
      data="NaN:NaN:NaN"
      title="Обновлен"
    />
    <Memo(wrappedComponent)
      data={5}
      title="Forks"
    />
    <Memo(wrappedComponent)
      data={5}
      title="Watchers"
    />
    <Memo(wrappedComponent)
      data={1}
      title="Open Issues"
    />
    <a
      className="btn btn-primary ml-auto mt-auto"
      href="https://google.com"
      target="_blank"
    >
      Перейти к репозиторию
    </a>
  </div>
</div>
`;
